yunwoo/programmers-1/명예의-전당-(1).js

```diff
@@ -0,0 +1,25 @@
+function solution(k, score) {
+  const result = [];
+  const hallOfFame = [];
+
+  for (const num of score) {
+    if (hallOfFame.length === k) {
+      hallOfFame[k - 1] = Math.max(hallOfFame.at(-1), num);
+    } else {
+      hallOfFame.push(num);
+    }
+
+    hallOfFame.sort((a, b) => b - a);
+    result.push(hallOfFame[hallOfFame.length - 1]);
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(k, score) {
  const hallOfFame = [];

  return score.reduce((acc, cur) => {
    hallOfFame.push(cur);
    hallOfFame.sort((a, b) => b - a)
      
    hallOfFame.length > k && hallOfFame.pop();

    return [...acc, hallOfFame.at(-1)];
  }, []);
}
```
- 풀이하느라 고생하셨습니다!~
- 아래와 같은 reduce, sort, push, pop 메서드를 활용해서도 풀이할 수 있을 것 같아요!! 참고해주세요

Reply:
- 와 코드가 훨씬 깔끔하네요 조건문도 없고!! 좋은 참고코드 감사합니다 🙇‍♂️